spicy/src/rt/base64.cc

```diff
@@ -61,6 +60,8 @@ hilti::rt::Bytes Stream::decode(const hilti::rt::Bytes& data) {
     if ( ! _state )
         throw Base64Error("decoding already finished");
 
+    base64_init_decodestate(&_state->dstate);
```

Comment:
This seems to not be what we want as it breaks the BTests. Not doing this leads to what looks like junk in the decoder stream.

Reply:
That resets the state, so that the "multiple calls" will presumably be breaking. If it doesn't work without that, we have a bug somewhere.

Reply:
I removed this again. The behavior I was attempting to "fix" was due to me using the base64 API incorrectly, in particular I was passing incomplete and unpadded data which lead to bytes I was not expecting to appear in the output. I fixed the test now.

I also attempted to upgrade libb64 to https://github.com/libb64/libb64's latest release (we are using 1.2.0 which is 10 years old while the latest release from the github fork is just over 2 years old). This failed since the github version changed its behavior.